diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/CheckInverseMethodRule.kt

```diff
@@ -35,9 +38,13 @@ class CheckInverseMethodRule(private val configRules: List<RulesConfig>) : Rule(
     }
 
     private fun checkCallExpressionName(node: ASTNode) {
-        if (node.treeParent.treePrev != null &&
-            node.treeParent.treePrev.elementType == OPERATION_REFERENCE &&
-            node.treeParent.treePrev.text == "!") {
+        val operationRef = node
+                .treeParent
+                .siblings(forward = false)
+                .takeWhile { it.elementType in intermediateTokens }
+                .filter { it.elementType == OPERATION_REFERENCE }
```

Comment:
```suggestion
                .firstOrNull { it.elementType == OPERATION_REFERENCE }
```
instead of creating a new list?